### PR TITLE
Support stable HTTP semconv

### PR DESCRIPTION
**Pull request by trask**
Depends on https://github.com/Azure/azure-sdk-for-java/pull/38276

### CODE REVIEW

File: smoke-tests/apps/Cassandra/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CassandraTest.java [hunk 1]
```diff
@@ -45,7 +45,7 @@ void cassandra() throws Exception {
     assertThat(telemetry.rdd1.getName()).isEqualTo("SELECT test.test");
     assertThat(telemetry.rdd1.getData()).isEqualTo("select * from test.test");
     assertThat(telemetry.rdd1.getType()).isEqualTo("cassandra");
-    assertThat(telemetry.rdd1.getTarget()).matches("dependency[0-9]+");
+    assertThat(telemetry.rdd1.getTarget()).matches("[0-9.]+");
```
Comment (trask) [inline smoke-tests/apps/Cassandra/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CassandraTest.java:48]: this feels like a regression, I think we need the upstream lettuce after all, can you update the upstream PR to implement both Network and Server attributes getter?

Comment (heyams) [reply smoke-tests/apps/Cassandra/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CassandraTest.java:48]: server is green.. i can add network.

**Comment by heyams** [comment]
smoke tests are green. will use the new exporter release to pass the rest.

**Review by heyams — APPROVED** [review]

File: agent/agent-tooling/build.gradle.kts [hunk 2]
```diff
@@ -21,9 +21,11 @@ dependencies {
   implementation(project(":agent:agent-profiler:agent-diagnostics"))
   implementation(project(":etw:java"))
 
-  implementation("com.azure:azure-monitor-opentelemetry-exporter:1.0.0-beta.15")
+  implementation("com.azure:azure-monitor-opentelemetry-exporter:1.0.0-beta.16")
   compileOnly("io.opentelemetry.javaagent:opentelemetry-javaagent-bootstrap")
-  compileOnly("io.opentelemetry.javaagent:opentelemetry-javaagent-tooling")
+  compileOnly("io.opentelemetry.javaagent:opentelemetry-javaagent-tooling") {
+    exclude("io.opentelemetry", "opentelemetry-exporter-prometheus")
+  }
```
Comment (trask) [inline agent/agent-tooling/build.gradle.kts:28]: can revert

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java [hunk 3]
```diff
@@ -21,6 +21,8 @@ public Map<String, String> apply(ConfigProperties otelConfig) {
     Configuration configuration = FirstEntryPoint.getConfiguration();
 
     Map<String, String> properties = new HashMap<>();
+    properties.put("otel.semconv-stability.opt-in", "http,jvm");
+
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java:25]: can revert

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/SamplingOverrides.java [hunk 4]
```diff
@@ -135,10 +139,15 @@ private StrictMatcher(String key, String value) {
       this.value = value;
     }
 
+    @SuppressWarnings("SystemOut")
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/SamplingOverrides.java:142]:
```suggestion
```

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/SamplingOverrides.java [hunk 5]
```diff
@@ -170,14 +180,20 @@ private RegexpMatcher(String key, String value) {
       this.value = Pattern.compile(value);
     }
 
+    @SuppressWarnings("SystemOut")
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/SamplingOverrides.java:183]:
```suggestion
```

File: smoke-tests/apps/HttpHeaders/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpHeadersTest.java [hunk 6]
```diff
@@ -20,14 +20,14 @@
 @UseAgent
 abstract class HttpHeadersTest {
 
-  @RegisterExtension static final SmokeTestExtension testing = SmokeTestExtension.create();
+  @RegisterExtension static final SmokeTestExtension testing = SmokeTestExtension.builder().setSelfDiagnosticsLevel("DEBUG").build();
```
Comment (trask) [inline smoke-tests/apps/HttpHeaders/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpHeadersTest.java:23]:
```suggestion
  @RegisterExtension static final SmokeTestExtension testing = SmokeTestExtension.create();
```

File: smoke-tests/apps/HttpHeaders/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpHeadersTest.java [hunk 7]
```diff
@@ -39,14 +39,14 @@ void testServerHeaders() throws Exception {
   @Test
   @TargetUri("/clientHeaders")
   void testClientHeaders() throws Exception {
-    Telemetry telemetry = testing.getTelemetry(1);
+    Telemetry telemetry = testing.getTelemetry(2);
```
Comment (trask) [inline smoke-tests/apps/HttpHeaders/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpHeadersTest.java:42]: is this change needed?

File: smoke-tests/apps/HttpPreaggregatedMetrics/src/main/java/com/microsoft/applicationinsights/smoketestapp/HttpPreaggregatedMetricsServlet.java [hunk 8]
```diff
@@ -40,9 +40,10 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
     }
   }
 
-  private void httpUrlConnection(String url) throws IOException {
+  private void httpUrlConnection(String url) throws IOException, InterruptedException {
     URL obj = new URL(url);
     HttpURLConnection connection = (HttpURLConnection) obj.openConnection();
+    Thread.sleep(200);
```
Comment (trask) [inline smoke-tests/apps/HttpPreaggregatedMetrics/src/main/java/com/microsoft/applicationinsights/smoketestapp/HttpPreaggregatedMetricsServlet.java:46]: I'd suggest try/catch InterruptedException just around only this one line

File: smoke-tests/apps/Lettuce/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/LettuceTest.java [hunk 9]
```diff
@@ -45,7 +45,7 @@ void lettuce() throws Exception {
     assertThat(telemetry.rdd1.getName()).isEqualTo("GET");
     assertThat(telemetry.rdd1.getData()).isEqualTo("GET test");
     assertThat(telemetry.rdd1.getType()).isEqualTo("redis");
-    assertThat(telemetry.rdd1.getTarget()).matches("dependency[0-9]+");
+    assertThat(telemetry.rdd1.getTarget()).matches("dependency[0-9.]+");
```
Comment (trask) [inline smoke-tests/apps/Lettuce/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/LettuceTest.java:48]: do you know why this change was needed?

File: smoke-tests/apps/HttpHeaders/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpHeadersTest.java (hunk 7 above)
Comment (heyams) [reply smoke-tests/apps/HttpHeaders/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpHeadersTest.java:42]: yes, it has two dependencies.

File: smoke-tests/apps/HttpHeaders/src/smokeTest/resources/applicationinsights-underscore.json [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+      "requestHeaders": [
+        "host"
+      ],
+      "responseHeaders": [
+        "abc_def"
+      ]
+    },
+    "captureHttpClientHeaders": {
+      "requestHeaders": [
+        "abc_def"
```
Comment (trask) [inline smoke-tests/apps/HttpHeaders/src/smokeTest/resources/applicationinsights-underscore.json:20]:
```suggestion
        "abc-def"
      ]
    },
    "captureHttpClientHeaders": {
      "requestHeaders": [
        "abc-def"
```

File: smoke-tests/apps/HttpHeaders/src/smokeTest/resources/applicationinsights-underscore.json [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  "sampling": {
+    "percentage": 100
+  },
+  "preview": {
+    "captureHttpServerHeaders": {
+      "requestHeaders": [
+        "host"
+      ],
+      "responseHeaders": [
+        "abc_def"
```
Comment (trask) [inline smoke-tests/apps/HttpHeaders/src/smokeTest/resources/applicationinsights-underscore.json:15]:
```suggestion
        "abc-def"
```

File: smoke-tests/apps/HttpHeaders/src/smokeTest/resources/applicationinsights-underscore.json (hunk 10 above)
Comment (trask) [inline smoke-tests/apps/HttpHeaders/src/smokeTest/resources/applicationinsights-underscore.json:20]:
```suggestion
        "abc-def"
```

File: smoke-tests/apps/HttpHeaders/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpHeadersUnderscoreTest.java [hunk 12]
```diff
@@ -17,17 +17,19 @@
...
 import org.junit.jupiter.api.extension.RegisterExtension;
 
-@UseAgent
+@UseAgent("applicationinsights-underscore.json")
 abstract class HttpHeadersTest {
 
-  @RegisterExtension static final SmokeTestExtension testing = SmokeTestExtension.create();
+  @RegisterExtension
+  static final SmokeTestExtension testing =
+      SmokeTestExtension.builder().setSelfDiagnosticsLevel("DEBUG").build();
```
Comment (trask) [inline smoke-tests/apps/HttpHeaders/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpHeadersUnderscoreTest.java:25]: can revert this when done

File: smoke-tests/apps/HttpHeaders/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpHeadersUnderscoreTest.java [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_8_OPENJ9;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.WILDFLY_13_JAVA_8;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.WILDFLY_13_JAVA_8_OPENJ9;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@UseAgent("applicationinsights-underscore.json")
+abstract class HttpHeadersUnderscoreTest {
```
Comment (trask) [inline smoke-tests/apps/HttpHeaders/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpHeadersUnderscoreTest.java:21]:
```suggestion
@UseAgent("applicationinsights-old-semconv.json")
abstract class HttpHeadersOldSemconvTest {
```

File: smoke-tests/apps/HttpHeaders/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpHeadersDashTest.java [hunk 14]
```diff
@@ -17,8 +17,8 @@
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
-@UseAgent
-abstract class HttpHeadersTest {
+@UseAgent("applicationinsights-dash.json")
+abstract class HttpHeadersDashTest {
```
Comment (trask) [inline smoke-tests/apps/HttpHeaders/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpHeadersDashTest.java:21]:
```suggestion
@UseAgent
abstract class HttpHeadersTest {
```

File: smoke-tests/apps/SpringBootAuto/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/SpringBootAutoTest.java [hunk 15]
```diff
@@ -32,7 +32,6 @@ void doMostBasicTest() throws Exception {
     assertThat(telemetry.rdEnvelope.getTags())
         .hasEntrySatisfying(
             "ai.user.userAgent", v -> assertThat(v).startsWith("Apache-HttpClient/"));
-    assertThat(telemetry.rdEnvelope.getTags()).containsKey("ai.location.ip");
```
[the remark this answers is not in the document]
Comment (heyams) [reply smoke-tests/apps/SpringBootAuto/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/SpringBootAutoTest.java:35]: String locationIp = getStableOrOldAttribute(attributes, SemanticAttributes.CLIENT_ADDRESS, // SemanticAttributes.HTTP_CLIENT_IP);
        if (locationIp == null) {
            // only use net.peer.ip if http.client_ip is not available
            locationIp = attributes.get(SemanticAttributes.NET_SOCK_PEER_ADDR);
        }

looks like a bug.. should we use SERVER_ADDRESS?

File: smoke-tests/apps/Cassandra/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CassandraTest.java [hunk 16]
```diff
@@ -45,7 +45,7 @@ void cassandra() throws Exception {
     assertThat(telemetry.rdd1.getName()).isEqualTo("SELECT test.test");
     assertThat(telemetry.rdd1.getData()).isEqualTo("select * from test.test");
     assertThat(telemetry.rdd1.getType()).isEqualTo("cassandra");
-    assertThat(telemetry.rdd1.getTarget()).matches("dependency[0-9]+");
+    assertThat(telemetry.rdd1.getTarget()).matches("cassandra[0-9.]?");
```
[the remark this answers is not in the document]
Comment (heyams) [reply smoke-tests/apps/Cassandra/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CassandraTest.java:48]: might need to fix upstream.

File: smoke-tests/apps/ConnectionStringOverrides/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/ConnectionStringOverridesTest.java [hunk 17]
```diff
@@ -99,9 +99,9 @@ private static void testApp(String iKey) throws Exception {
     // and metrics too
 
     List<Envelope> clientMetrics =
-        testing.mockedIngestion.waitForMetricItems("http.client.duration", 1);
+        testing.mockedIngestion.waitForMetricItems("http.client.request.duration", 1);
     List<Envelope> serverMetrics =
-        testing.mockedIngestion.waitForMetricItems("http.server.duration", 1);
+        testing.mockedIngestion.waitForMetricItems("http.server.request.duration", 1);
```
[the remark this answers is not in the document]
Comment (heyams) [reply smoke-tests/apps/ConnectionStringOverrides/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/ConnectionStringOverridesTest.java:104]: that is "_MS.MetricId":"requests/duration"

[the remark this answers is not in the document]
Comment (heyams) [reply smoke-tests/apps/ConnectionStringOverrides/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/ConnectionStringOverridesTest.java:104]: let's chat.. not sure i understand the confusion.. metric name is straight from the MetricData.

[the remark this answers is not in the document]
Comment (heyams) [reply smoke-tests/apps/ConnectionStringOverrides/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/ConnectionStringOverridesTest.java:104]: mdm understand metricId not name.

File: smoke-tests/apps/Cassandra/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CassandraTest.java (hunk 16 above)
[the remark this answers is not in the document]
Comment (heyams) [reply smoke-tests/apps/Cassandra/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CassandraTest.java]: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10357

File: smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-old-semconv.json [hunk 18]
```diff
@@ -0,0 +1,114 @@
...
+  },
+  "preview": {
+    "processors": [
+      {
+        "type": "attribute",
+        "actions": [
+          {
+            "key": "http.method",
+            "value": "GET",
+            "action": "insert"
```
Comment (trask) [inline smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights-old-semconv.json:17]: I didn't follow the change here from the original, which was

```
          {
            "key": "attribute1",
            "value": "testValue",
            "action": "insert"
          },
```

File: agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/processors/TelemetryProcessorMaskingTest.java [hunk 19]
```diff
@@ -166,7 +166,7 @@ public long getStartEpochNanos() {
     @Override
     public Attributes getAttributes() {
       return Attributes.builder()
-          .put("http.status_code", 200L)
+          .put("http.response.status_code", 200L)
           .put("http.url", httpUrl)
           .put("http.method", "GET")
```
Comment (trask) [inline agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/processors/TelemetryProcessorMaskingTest.java:171]: I think we want these to be the new attributes, to verify that the old `http.url` based telemetry processor still works
```suggestion
          .put("url.full", httpUrl)
          .put("http.request.method", "GET")
```

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/SamplingOverrides.java [hunk 20]
```diff
@@ -244,9 +265,40 @@ private String get() {
     }
   }
 
+  private static boolean getHttpUrlKeyOldOrStableSemconv(AttributeKey<String> key) {
+    String keyString = key.getKey();
+    return keyString.equals(SemanticAttributes.HTTP_URL.getKey())
+        || keyString.equals(SemanticAttributes.URL_PATH.getKey())
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/SamplingOverrides.java:271]: `http.path` is only partial url
```suggestion
```

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java [hunk 21]
```diff
@@ -337,6 +337,9 @@ private static Set<Feature> initStatsbeatFeatureSet(Configuration config) {
     if (!config.preview.instrumentation.jaxrsAnnotations.enabled) {
       featureList.add(Feature.JAXRS_ANNOTATIONS_DISABLED);
     }
+    if (!config.preview.instrumentation.pekko.enabled) {
+      featureList.add(Feature.PEKKO_DISABLED);
```
Comment (jeanbisutti) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java:341]: 👍

File: smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights.json [hunk 22]
```diff
@@ -78,8 +78,8 @@
         "type": "attribute",
         "actions": [
           {
-            "key": "http.url",
-            "pattern": "https?://[^/]+(?<httpPath>/[^?]*)",
+            "key": "url.path",
+            "pattern": "(?<httpPath>/[^?]*)",
```
Comment (trask) [inline smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights.json:82]: the equivalent of `http.url` in the new semconv is `url.full`

File: smoke-tests/apps/TelemetryProcessors/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/TelemetryProcessorDeleteLogAttribute.java [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+  static final SmokeTestExtension testing =
+      SmokeTestExtension.builder().setSelfDiagnosticsLevel("DEBUG").build();
+
+  @Test
+  @TargetUri("/delete-existing-log-attribute")
+  void deleteExistingLogAttribute() throws Exception {
+    Telemetry telemetry = testing.getTelemetry(0);
+    assertThat(telemetry.rd.getName())
+        .isEqualTo("GET /TelemetryProcessors/delete-existing-log-attribute");
+    assertThat(telemetry.rd.getProperties().get("toBeDeletedAttributeKey")).isNull();
```
Comment (trask) [inline smoke-tests/apps/TelemetryProcessors/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/TelemetryProcessorDeleteLogAttribute.java:35]: can you assert against all properties, so we can be sure it didn't get transformed into something else?

File: smoke-tests/apps/TelemetryProcessors/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/TelemetryProcessorDeleteLogAttribute.java [hunk 24]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@UseAgent("applicationinsights-delete-existing-attribute.json")
+abstract class TelemetryProcessorDeleteLogAttribute {
+
+  @RegisterExtension
+  static final SmokeTestExtension testing =
+      SmokeTestExtension.builder().setSelfDiagnosticsLevel("DEBUG").build();
```
Comment (trask) [inline smoke-tests/apps/TelemetryProcessors/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/TelemetryProcessorDeleteLogAttribute.java:27]:
```suggestion
      SmokeTestExtension.create();
```

File: smoke-tests/apps/Cassandra/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CassandraTest.java [hunk 25]
```diff
@@ -45,7 +45,9 @@ void cassandra() throws Exception {
     assertThat(telemetry.rdd1.getName()).isEqualTo("SELECT test.test");
     assertThat(telemetry.rdd1.getData()).isEqualTo("select * from test.test");
     assertThat(telemetry.rdd1.getType()).isEqualTo("cassandra");
-    assertThat(telemetry.rdd1.getTarget()).matches("dependency[0-9]+");
+    //    assertThat(telemetry.rdd1.getTarget()).matches("dependency[0-9]+"); // need to look into
+    // upstream
+    assertThat(telemetry.rdd1.getTarget()).matches("cassandra");
```
Comment (trask) [inline smoke-tests/apps/Cassandra/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CassandraTest.java:50]: this should be fixed by `2.1.0`?

File: smoke-tests/apps/SpringBootAuto/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/SpringBootAutoTest.java [hunk 26]
```diff
@@ -32,7 +32,8 @@ void doMostBasicTest() throws Exception {
...
+    //        assertThat(telemetry.rdEnvelope.getTags()).containsKey("ai.location.ip"); // TODO
+    // waiting for the fix to be released in upstream
```
Comment (trask) [inline smoke-tests/apps/SpringBootAuto/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/SpringBootAutoTest.java:36]: this should be fixed by `2.1.0`?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java [hunk 27]
```diff
@@ -270,6 +274,92 @@ private static void overlayConfiguration(
...
+    } else if (oldAttributeKey.equals(SemanticAttributes.NETWORK_PEER_PORT.getKey())) {
+      return SemanticAttributes.SERVER_PORT.getKey();
+    }
+
+    // HTTP server span attributes
+    // http.target is handled via LazyHttpTarget
+    if (oldAttributeKey.equals(SemanticAttributes.HTTP_SCHEME.getKey())) {
+      return SemanticAttributes.URL_SCHEME.getKey();
+    } else if (oldAttributeKey.equals(SemanticAttributes.HTTP_RESEND_COUNT.getKey())) {
+      return SemanticAttributes.CLIENT_ADDRESS.getKey();
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:356]: this doesn't look right to me?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java [hunk 28]
```diff
@@ -270,6 +274,92 @@ private static void overlayConfiguration(
...
+    } else if (oldAttributeKey.equals(SemanticAttributes.NET_SOCK_PEER_ADDR.getKey())) {
+      return SemanticAttributes.NETWORK_PEER_ADDRESS.getKey();
+    } else if (oldAttributeKey.equals(SemanticAttributes.NET_SOCK_PEER_PORT.getKey())) {
+      return SemanticAttributes.NETWORK_PEER_PORT.getKey();
+    }
+
+    // HTTP client span attributes
+    // http.url is handled via LazyHttpUrl
+    if (oldAttributeKey.equals(SemanticAttributes.HTTP_RESEND_COUNT.getKey())) {
+      return "http.request.resend_count";
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:344]: is there no constant for this?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java [hunk 29]
```diff
@@ -270,6 +274,92 @@ private static void overlayConfiguration(
...
+    } else if (oldAttributeKey.equals(SemanticAttributes.NET_PEER_NAME.getKey())) {
+      return SemanticAttributes.SERVER_ADDRESS.getKey();
+    } else if (oldAttributeKey.equals(SemanticAttributes.NETWORK_PEER_PORT.getKey())) {
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:347]:
```suggestion
    } else if (oldAttributeKey.equals(SemanticAttributes.NET_PEER_PORT.getKey())) {
```

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java [hunk 30]
```diff
@@ -270,6 +274,92 @@ private static void overlayConfiguration(
...
+    supportTelemetryProcessorsOldSemConv(config);
+  }
+
+  private static void supportSamplingOverridesOldSemConv(SamplingOverride override) {
+    for (Configuration.SamplingOverrideAttribute attribute : override.attributes) {
+      attribute.key = mapAttributeKey(attribute.key);
+    }
+  }
+
+  private static void supportTelemetryProcessorsOldSemConv(Configuration config) {
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:286]: can we emit a warning when we find that the user is using one of the old attributes?

File: smoke-tests/apps/TelemetryProcessors/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/TelemetryProcessorDeleteLogAttribute.java [hunk 31]
```diff
@@ -0,0 +1,72 @@
...
+        .isEqualTo("GET /TelemetryProcessors/delete-existing-log-attribute");
+    assertThat(telemetry.rd.getProperties().get("toBeDeletedAttributeKey")).isNull();
+
+    List<MessageData> logs = testing.mockedIngestion.getTelemetryDataByType("MessageData");
+    logs.stream()
+        .filter(log -> log.getMessage().contains("custom property from MDC"))
+        .forEach(
+            log -> {
+              assertThat(log.getProperties().get("toBeDeletedAttributeKey")).isNull();
+            });
```
[the remark this answers is not in the document]
Comment (heyams) [reply smoke-tests/apps/TelemetryProcessors/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/TelemetryProcessorDeleteLogAttribute.java:43]: there are more than one though

image

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java (hunk 27 above)
Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:356]: should be client_ip.

File: smoke-tests/apps/SpringBootAuto/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/SpringBootAutoTest.java (hunk 26 above)
Comment (heyams) [reply smoke-tests/apps/SpringBootAuto/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/SpringBootAutoTest.java]: yup add a todo

File: smoke-tests/apps/Cassandra/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CassandraTest.java (hunk 25 above)
Comment (heyams) [reply smoke-tests/apps/Cassandra/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/CassandraTest.java]: yup, update todo

File: smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights.json (hunk 22 above)
Comment (heyams) [reply smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights.json:82]: this is a server span. url.full is only for client span

File: smoke-tests/apps/TelemetryProcessors/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/TelemetryProcessorDeleteLogAttribute.java (hunk 31 above)
[the remark this answers is not in the document]
Comment (trask) [reply smoke-tests/apps/TelemetryProcessors/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/TelemetryProcessorDeleteLogAttribute.java:46]: check how other smoke tests work

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java (hunk 28 above)
Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java]: I see constant being used in upstream repo: https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-java-instrumentation%20http_request_resend_count&type=code

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java [hunk 32]
```diff
@@ -270,6 +274,102 @@ private static void overlayConfiguration(
...
+      result = SemanticAttributes.SERVER_ADDRESS.getKey();
+    } else if (oldAttributeKey.equals(SemanticAttributes.NET_HOST_PORT.getKey())) {
+      result = SemanticAttributes.SERVER_PORT.getKey();
+    }
+
+    if (result == null) {
+      result = oldAttributeKey;
+    } else {
+      configurationLogger.warn(
+          "\"{}\" has been deprecated and replaced with \"{}}\" since 3.5 GA.",
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:368]:
```suggestion
          "\"{}\" has been deprecated and replaced with \"{}\" since 3.5.0 GA.",
```

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java (hunk 28 above)
Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java]: we used the copy from the standalone exporter. this constant is not available.

Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java]: // this is a copy of io.opentelemetry.semconv.SemanticAttributes
// because the module that contains that class is not stable,
// so don't want to take a dependency on it

when can we use the upstream copy?

Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java]: we only need to use the copy in the azure sdk repo

File: smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights.json (hunk 22 above)
Comment (trask) [reply smoke-tests/apps/TelemetryProcessors/src/smokeTest/resources/applicationinsights.json:82]: oh yes, makes sense

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java (hunk 28 above)
Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java]: can you be more specific about it? SemanticAttributes in https://mvnrepository.com/artifact/io.opentelemetry/opentelemetry-semconv/1.30.1-alpha is deprecated.

Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java]: import io.opentelemetry.semconv.SemanticAttributes; 1.21 missing some other attributes. 1.30 is deprecated.
